File: scala-finch/src/main/scala/org/openapitools/models/ResponseTimeFilterProperties.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import traveltimeplatform._
import org.openapitools.models.ResponseDistanceBreakdownItem
import org.openapitools.models.ResponseFares
import org.openapitools.models.ResponseRoute
import scala.collection.immutable.Seq

/**
 * 
 * @param travelUnderscoretime 
 * @param distance 
 * @param distanceUnderscorebreakdown 
 * @param fares 
 * @param route 
 */
case class ResponseTimeFilterProperties(travelUnderscoretime: Option[Int],
                distance: Option[Int],
                distanceUnderscorebreakdown: Option[Seq[ResponseDistanceBreakdownItem]],
                fares: Option[ResponseFares],
                route: Option[ResponseRoute]
                )

object ResponseTimeFilterProperties {
    /**
     * Creates the codec for converting ResponseTimeFilterProperties from and to JSON.
     */
    implicit val decoder: Decoder[ResponseTimeFilterProperties] = deriveDecoder
    implicit val encoder: ObjectEncoder[ResponseTimeFilterProperties] = deriveEncoder
}

File: fsharp-functions/OpenAPI/src/model/ResponseTimeFilterProperties.fs
namespace OpenAPI.Model

open System
open System.Collections.Generic
open Newtonsoft.Json
open OpenAPI.Model.ResponseDistanceBreakdownItem
open OpenAPI.Model.ResponseFares
open OpenAPI.Model.ResponseRoute

module ResponseTimeFilterProperties = 

  //#region ResponseTimeFilterProperties

  [<CLIMutable>]
  type ResponseTimeFilterProperties = {
    [<JsonProperty(PropertyName = "travel_time")>]
    TravelTime : int;
    [<JsonProperty(PropertyName = "distance")>]
    Distance : int;
    [<JsonProperty(PropertyName = "distance_breakdown")>]
    DistanceBreakdown : ResponseDistanceBreakdownItem[];
    [<JsonProperty(PropertyName = "fares")>]
    Fares : ResponseFares;
    [<JsonProperty(PropertyName = "route")>]
    Route : ResponseRoute;
  }
  
  //#endregion
  